File: LaibaYousaf Assignment Traffic_Light_Controller_/Code/traffic_light.py
class TrafficLight:
    def __init__(self):
        self.state = "Red"

    def transition(self):
        if self.state == "Red":
            self.state = "Green"
        elif self.state == "Green":
            self.state = "Yellow"
        elif self.state == "Yellow":
            self.state = "Red"

    def current_state(self):
        return self.state

File: LaibaYousaf Assignment Traffic_Light_Controller_/Code/traffic_light.c
#include <stdio.h>

typedef enum { RED, GREEN, YELLOW } State;

State next_state(State current) {
    switch (current) {
        case RED: return GREEN;
        case GREEN: return YELLOW;
        case YELLOW: return RED;
    }
}

int main() {
    State current = RED;
    State next = next_state(current);

    // Symbolic execution starts here
    printf("Current State: %d, Next State: %d\n", current, next);

    return 0;
}
